### PR TITLE
Introduce a standalone build of the haskell backend

### PR DESCRIPTION
**Pull request by ehildenb**
~Blocked on: https://github.com/runtimeverification/pyk/pull/442~
~Blocked on: https://github.com/runtimeverification/evm-semantics/pull/1831~

This standalone build is the first step towards executing the bchain conformance tests on the haskell backend directly.

- The `make build-haskell-standalone` target is added.
- The configuration needed is added to `kevm-pyk` in `config.py` and `kompile.py` to build this target.
- The `exec_run` method of `kevm-pyk` is adjusted to work correctly with configuration variables in KEVM.
- Tests of `kevm run --pyk ... [--backend llvm|haskell-standalone]` are added to make sure they work on CI.

### CODE REVIEW

File: kevm-pyk/src/kevm_pyk/kompile.py [hunk 1]
```diff
@@ -128,6 +141,7 @@ def kevm_kompile(
     )
 
     kompile: Kompile
+    haskell_binary = not (Kernel.get() == Kernel.DARWIN)
```
Comment (tothtamas28) [inline kevm-pyk/src/kevm_pyk/kompile.py:144]:
```suggestion
    kernel = Kernel.get()
    haskell_binary = kernel is not Kernel.DARWIN
```

File: kevm-pyk/src/kevm_pyk/kompile.py [hunk 2]
```diff
@@ -171,15 +186,6 @@ def _lib_ccopts() -> list[str]:
...
-    class Kernel(Enum):
-        LINUX = 'Linux'
-        DARWIN = 'Darwin'
-
-        @staticmethod
-        def get() -> Kernel:
-            uname = run_process(('uname', '-s'), pipe_stderr=True, logger=_LOGGER).stdout.strip()
-            return Kernel(uname)
-
     kernel = Kernel.get()
```
Comment (tothtamas28) [inline kevm-pyk/src/kevm_pyk/kompile.py:189]: This can then be removed.

**Review by tothtamas28 — APPROVED** [review]

File: kevm-pyk/src/kevm_pyk/kompile.py (hunk 2 above)
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/kompile.py]: Done.

File: kevm-pyk/src/kevm_pyk/kompile.py (hunk 1 above)
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/kompile.py]: Done.

**Pull request merged**